source/MissionAction.cpp

```diff
@@ -366,3 +371,35 @@ int64_t MissionAction::Payment() const noexcept
 {
 	return action.Payment();
 }
+
+
+
+shared_ptr<ConditionSet> MissionAction::GetToDisplay(const DataNode &node)
+{
+	shared_ptr<ConditionSet> conditions;
+	for(auto &child : node)
+		if(child.Size() != 2 || child.Token(0) != "to" || child.Token(1) != "display" || !child.HasChildren())
+			child.PrintTrace("Ignoring unrecognized dialog token");
+		else if(!conditions)
+			conditions = make_shared<ConditionSet>(child);
+		else
+			conditions->Load(child);
+	return conditions;
+}
+
+
+
+void MissionAction::LoadDialog(const DataNode &node)
+{
+	// Parse the "dialog phrase whatever" and "dialog whatever" lines:
+	if(node.Size() == 3 && node.Token(1) == "phrase")
+		dialog.emplace_back("${" + node.Token(2) + "}", nullptr);
+	else if(node.Size() == 2)
+		dialog.emplace_back(node.Token(1), nullptr);
+	// Parse dialog in children:
+	for(auto &child : node)
+		if(child.Size() == 2 && child.Token(0) == "phrase")
+			dialog.emplace_back("${" + child.Token(1) + "}", GetToDisplay(child));
+		else
+			dialog.emplace_back(child.Token(0), GetToDisplay(child));
+}
```

Comment:
The original code also had some PrintTraces for invalid syntax. Can we also verify the syntax here? (Or will we never have any invalid syntax after this change?)

Reply:
I have updated it to verify the syntax.